### ISSUE TITLE
Allow selecting multiple tags to filter the tasks view

### ISSUE THREAD

**Issue by tkdchen** [getting-things-gnome/gtg#123]
Usually, a task has several tags. It would be much convenient to select more than one tags to filter tasks.

**Comment by izidormatusov** [comment]
You can do that with search currently. Selecting multiple tags provides quite challenges to create meaningful UI. Currently tags work as a category metaphor: are you in active tasks? Are you in `@school` tasks?

When you provide multi selection, then you  start going into drilling down and switching from `@school` & `@math` to `@home` is challenging.

Please provide UI mockups how it should work.

**Comment by tkdchen** [comment]
My use case is simple. After I select one tag,
![select-one-tag](https://cloud.githubusercontent.com/assets/919706/7984931/5888d8a0-0aff-11e5-9deb-ac7f69f53940.png)

then, I would like to see only tasks that have both `@Bug` and `@work` by selecting `work`.

![select-two-tags](https://cloud.githubusercontent.com/assets/919706/7985012/1a1f751e-0b00-11e5-8fac-6951f073afb5.png)

**Comment by nekohayo** [comment]
I can see the appeal of this for someone to be able to say "Today I want to view these three topic tags: contracts, accounting, taxes; but those are totally separate categories normally, they are not parent/children of each other". Basically be able to create your own focused view tailored to each day. It would feel more "natural" to be able to do this with clicking rather than creating a custom search with syntax like `@contracts !or @accounting !or @taxes`.

From a technical perspective, this could be simpler to implement than it seems, as a semi-hidden feature: you don't necessarily need to expose a custom UI for this (that would be a pain in the butt), simply support ctrl+click to add/remove tags to the current filtering selection.

* Ctrl+click on a tag in the sidebar adds (or removes) it to the current selection of tags affecting the filtering of the view. The search/filter refreshes each time that happens.
* If no tag was selected before (i.e. "All tasks" was selected), ctrl+click behaves as a regular click (it replaces the current "All tasks" selection)
* If multiple tags are selected, creating a new task uses those tags when creating the new task (it currently does that for single tag selections)
* A regular click (without the Ctrl modifier) unselects the tags and selects only the new single selected tag. The potentially tricky part to take into account is making sure that the search/filtering happens only once in this particular case (i.e. on selecting the new tag, not for every deselected tag as part of that click).

Ctrl selections would be enough; I don't think it makes sense to have Shift (range) selections because of the alphabetic sorting and because at that point the user might just as well use hierarchical tags.

Without needing to implement a whole GUI metaphor for it (other than support for the Ctrl keyboard modifier), I think this could be an easy-enough enhancement request to be considered "low-hanging-fruit".
